src/genType.ts

```diff
@@ -69,28 +43,14 @@ export function getLoaderTypeKey(resourceConfig: ResourceConfig, resourcePath: R
             )}`;
         }
 
-        return `{|
-            ...$Diff<${resourceArgs}, {
-                ${resourceConfig.batchKey}: $PropertyType<${resourceArgs}, '${resourceConfig.batchKey}'>
-            }>,
-            ...{| ${newKeyType} |}
-        |}`;
+        return `Omit<${resourceArgs}, '${resourceConfig.batchKey}'> & { ${newKeyType} }`;
     }
 
     return resourceArgs;
 }
 
 export function getLoaderTypeVal(resourceConfig: ResourceConfig, resourcePath: ReadonlyArray<string>) {
-    // TODO: We assume that the resource accepts a single dict argument. Let's
-    // make this configurable to handle resources that use seperate arguments.
```

Comment:
is this comment not still relevant?

Reply:
I think I just got a little overzealous assuming this TODO was not going to happen. I can add it back.